### PR TITLE
Make sure vault integration works with PSPs

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
- Update server-acl-init PSP to have access to mount emptyDir volumes for the vault injector
- Scale up gke cluster so accommodate for everything we're installing in the vault test

How I've tested this PR:
acceptance tests: https://app.circleci.com/pipelines/github/hashicorp/consul-k8s/4015/workflows/87c78757-4c96-4641-a2f0-d535bbe567fc/jobs/36509

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/server-acl-init-podsecuritypolicy.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ spec:
   # Allow core volume types.
   volumes:
     - 'secret'
+    - 'emptyDir'
```
Comment (thisisnotashwin) [inline charts/consul/templates/server-acl-init-podsecuritypolicy.yaml:21]: GENIUS! pure unadulterated genius.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ IMPROVEMENTS:
   * Allow customization of `terminationGracePeriodSeconds` on the ingress gateways. [[GH-947](https://github.com/hashicorp/consul-k8s/pull/947)]
   * Support `ui.dashboardURLTemplates.service` value for setting [dashboard URL templates](https://www.consul.io/docs/agent/options#ui_config_dashboard_url_templates_service). [[GH-937](https://github.com/hashicorp/consul-k8s/pull/937)]
   * Allow using dash-separated names for config entries when using `kubectl`. [[GH-965](https://github.com/hashicorp/consul-k8s/pull/965)]
+  * Support Pod Security Policies with Vault integration. [[GH-985](https://github.com/hashicorp/consul-k8s/pull/985)]
```
Comment (thisisnotashwin) [inline CHANGELOG.md:13]: 💥

**Review by thisisnotashwin — APPROVED** [review]
🪄 This is a very very good find! Devil in the details and all that.

File: charts/consul/templates/server-acl-init-podsecuritypolicy.yaml (hunk 1 above)
Comment (kschoche) [reply charts/consul/templates/server-acl-init-podsecuritypolicy.yaml:21]: Nice!

**Review by kschoche — APPROVED** [review]
Woohoo thanks!

**Pull request merged**